Add life burst icon

Deardrops opened the pull request:
TODO: 

- [x] 定位 life burst icon

webxoss committed:
add menu

webxoss committed:
improve style

webxoss committed:
add basic deck operation

webxoss committed:
rearrange construct

Deardrops committed:
add life burst icon

webxoss committed:
disable ok button for invalid input

webxoss commented on src/components/Cell.vue:
```diff
@@ -120,6 +122,7 @@ export default {
   <router-link :to="detailRoute">
     <div :class="[$style.cell]">
       <thumbnail :class="[$style.thumbnail, $color[card.color]]" :pid="card.pid"></thumbnail>
+      <icon :class="$style.burst" name="burst"/>
```
burst icon 要不要作为 thumbnail 的一部分呢？
@Deardrops @gugugupan

Deardrops replied:
在Detail页面上的thumbnail不需要这个icon，
放在Cell组件上好定位一些，大概
我的想法是放到Cell上

Deardrops committed:
Merge branch 'master' into dev

Deardrops committed:
add life burst Icon

Deardrops committed:
merge master

Deardrops commented:
@gugugupan 能帮忙定位一下这个icon么？相对定位到缩略图的下部，可以超出一点点

webxoss commented:
放到缩略图下面？全部在下面会很难看吧，一半在里面的话，颜色怎么处理？

Deardrops committed:
position life burst icon

style life burst icon

position by percentage

hexagon icon border

simplefy hexagon icon

reordering css

Deardrops commented:
团子三天前就敏锐的意识到了这个问题，

~~把icon放到卡名那一行的最后（也就是右上角）试试效果吧~~

现在的效果：

![6e s 69an62qc9 boc9b](https://cloud.githubusercontent.com/assets/9463871/22616478/a98093b8-eae8-11e6-9430-7f75f1a82b4d.png)

Deardrops commented on src/components/Cell.vue:
```diff
@@ -161,15 +161,18 @@ export default {
+  transform: scale(1.5);
+  background-color: currentColor;
+}
+.burst use {
```
.burst>use {

webxoss replied:
不要企图改变组件内部元素的样式，你不应该知道组件内部的 DOM 结构

```css
.burst {
  color: #fff;
}
```
应该就可以了

Deardrops replied:
background-color: currentColor 背景颜色设置为卡牌的颜色，即依赖于color，
如果这样做，将icon组件内的颜色 设置为白色，这里定义同时也修改了color，
结果导致icon和背景颜色相同，怎么解决这个冲突

webxoss replied:
图标容器 color 为卡片颜色，里面图标 color 为白色不就好了吗？

webxoss commented on src/components/Cell.vue:
```diff
@@ -149,6 +158,22 @@ export default {
+  left: 2.75em;
+  z-index: 1;
+  border-radius: 50%;
+  transform: scale(1.5);
```
这定位23333
用 wrapper 包住缩略图和图标，相对定位，图标绝对定位， top100%，left50%，transform向左向上各偏移50% 就好了

ps: icon 组件的大小用 font-size 就好

webxoss commented on src/components/DeckButtons.vue:
```diff
@@ -0,0 +1,142 @@
+  height: 3rem;
+  font-size: 1.3em;
+}
+</style>
```
不小心弄进来的吧，删掉就好

webxoss commented on src/css/vars.css:
```diff
@@ -4,9 +4,9 @@
-  --z-modal: 3;
+  --z-header: 10;
+  --z-header-menu: 20;
+  --z-modal: 30;
```
其实有没有更好的管理方法？

Deardrops commented:
当前效果：
![w atu3solr zs5 b_smoe](https://cloud.githubusercontent.com/assets/9463871/22626983/edf17fec-ebf3-11e6-8094-276c5aa24faf.png)

webxoss commented on src/components/Cell.vue:
```diff
@@ -149,6 +163,25 @@ export default {
+  top: 0%;
+  left: 50%;
+  transform: translate(-50%,-50%);
+  width: 1em;
```
[+] `height: 1em;`

webxoss commented on src/components/Cell.vue:
```diff
@@ -149,6 +163,25 @@ export default {
+  left: 50%;
+  transform: translate(-50%,-50%);
+  width: 1em;
+  font-size: 1.5em;
```
[+] `line-height: 1;`

webxoss commented on src/components/Cell.vue:
```diff
@@ -149,6 +163,25 @@ export default {
+  transform: scale(0.8);
+}
+.hexagon {
+  position: relative;
```
absolute

webxoss commented on src/components/Cell.vue:
```diff
@@ -149,6 +163,25 @@ export default {
+  height: 0px;
+}
+.burst {
+  color: #fff;
```
放下面吧，

一般把 display, position 之类重要的放前面

webxoss commented on src/components/Cell.vue:
```diff
@@ -149,6 +163,25 @@ export default {
 
+.wrapper {
+  position: relative;
+  height: 0px;
```
0 不要单位

webxoss reviewed (approved):
这个pr历史有点诡异，仔细 rebase 一下

我先 approve 睡觉了，弄好了自己合吧

Deardrops committed:
Merge branch 'master' into dev
